include/rmm/mr/device_memory_resource.hpp

```diff
@@ -107,6 +107,18 @@ class device_memory_resource {
    *---------------------------------------------------------------------------**/
   virtual bool supports_streams() const noexcept = 0;
 
+  /**---------------------------------------------------------------------------*
+   * @brief Queries if the the amount of free and total memory for the resource.
+   *
+   * @param freeSize the amount of free memory
+   * @param totalSize the total amount of memory
+   * @param stream the stream whose memory manager we want to retrieve
+   *
+   * @returns If the resource supports non-null streams
+   *---------------------------------------------------------------------------**/
+  void get_mem_info(size_t *freeSize, size_t *totalSize, cudaStream_t stream){
```

Comment:
I like this, I'd just make it such that the info is returned rather than using output parameters, e.g., 
```suggestion
  std::pair<std::size_t, std::size_t> get_mem_info(cudaStream_t stream){
```

or, could have a simple struct like:
```
struct mem_info{
   std::size_t free{};
   std::size_t available{};
};

  mem_info get_mem_info(cudaStream_t stream)
```

Reply:
i just used std::pair